### ISSUE TITLE
{edition} tags lose closing bracket with custom text

### ISSUE THREAD

**Issue by JohnnyGrey86** [Radarr/Radarr#9151]
### Is there an existing issue for this?

- [X] I have searched the existing open and closed issues

### Current Behavior

I fully acknowledge that I am using the edition tags outside their intended purpose. I am adding the custom text "4K" to my edition tags in order to force Plex to recognize files from this instance of Radarr as separate library items from my main instance. This text should appear before the real edition tags, if one is present. When one is not present, the closing curly brace is being left out of the file name, causing Plex to think the edition contains additional text from later on in the filename.

As an example, the following movie format:
`{imdb-{ImdbId}} {edition-4K {Edition Tags}} {[Custom Formats]}`
should result in the following, which is how it shows in the preview:
`{imdb-tt0066921} {edition-4K Ultimate Extended Edition} [Surround Sound x264]`
or, alternatively without a supplied edition (not shown in preview):
`{imdb-tt0066921} {edition-4K} [Surround Sound x264]`

However, when there is no edition in the grab, the closing curly brace is left out, as in:
`Tenet (2020) {imdb-tt6723592} {edition-4K [Hybrid][Remux-2160p]`

### Expected Behavior

Closing curly brace should be present given that _some_ text is being provided for the edition.

### Steps To Reproduce

1. Add custom text to within the {edition-} token file name format.
2. Let a grab finish downloading.
3. Let Radarr move the file and rename the file.
4. Observe file name is incorrect.

### Environment

```markdown
- OS: Unraid 6.11.5
- Radarr: 4.7.5.7809
- Docker Install: Yes
- Using Reverse Proxy: No
- Browser: Edge Version 116.0.1938.69
- Database: Sqlite 3.40.1
```

### What branch are you running?

Master

### Trace Logs? **Not Optional**

https://gist.github.com/JohnnyGrey86/0716bf2a949dd82f87bba45c767d189e

### Trace Logs have been provided as applicable. Reports will be closed if the required logs are not provided.

- [X] I have read and followed the steps in the wiki link above and provided the required trace logs - the logs contain `trace` - that are relevant and show this issue.

**Comment by UberNude** [comment]
I found a "fix" for this interaction. I'm not sure the reason but Radarr does not like when it just ends in 4K. To fix this you can insert a blank invisible character after 4K instead of a space. I'm not sure of any implications this can cause but it has worked on my windows server.

Here is your formatting with the invisible character after the 4K. 
`{imdb-{ImdbId}} {edition-4K‎‎ {Edition Tags}} {[Custom Formats]}`

When there are no edition tags this will result in this output.
`Tenet (2020) {imdb-tt6723592} {edition-4K‎ } [Hybrid][Remux-2160p]`

This will appear in Plex with the edition as 4K with the blank character but you can't tell visually.

You can find the invisible character yourself here, just copy and paste it.
https://www.editpad.org/tool/invisible-character

**Comment by yandy-r** [comment]
Same issue, as I'm trying to organize movies in the same /Movies directory. However, I could get away with adding quotes ("").

```
{edition-"4K {Edition Tags}"}
```

This does add a space if no **Edition Tags** are found, but at least I don't lose that information. When the bug gets fixed, I can go back and remove the quotes.

**Comment by Zerauskire** [comment]
I agree it would be nice to have a fix for this. I'm running in to the same issue. My current format is as follows:
`{Movie Title} ({Release Year}).{Quality Title}.{MediaInfo VideoCodec}.{MediaInfo AudioCodec}.{MediaInfo AudioChannels} {tmdb-{TmdbId}} {edition-{Edition Tags}}`

This results in the end of the file being "{edition-.mkv".

I have a suggestion for this. What if you were to create some sort of solution where you bind the surrounding text to the tag using some sort of tags before and after. Using my above format as an example lets say I wanted " {edition-{Edition Tags}}" to show up at the end of my file but if there's no value in the {Edition Tags} then i just want it to end with "{TmdbId}}.mkv"

So what you could implement is some method where you could format it like this:
`{Movie Title} ({Release Year}).{Quality Title}.{MediaInfo VideoCodec}.{MediaInfo AudioCodec}.{MediaInfo AudioChannels} {tmdb-{TmdbId}}< {edition-{Edition Tags}}>`

This way the code knows that if the recognized tag within the <> returns a null value then anything else within the <> is not added. Notice how I also included the space " " at the front as well so that we don't end up with a space at the end of the file. 

While "<" and ">" are illegal characters in a Windows filename they are not in MacOS and Linux so maybe you could use something else. I'm not sure. The only thing I know of that can't be in the file names of any of those is "\0" so if you're worried about people wanting to use "<" or ">" in their filenames then I suppose you could use "\0" instead. I don't know the best answer for deciding what character(s) to use for this but I think the solution itself is pretty straight forward and works not just for the Editions Tags but for any other tags that may result in a null value for that matter going forward.

**Comment by bl4ckswordsman** [comment]
I also have this issue and I'm currently using the quotes (" ") workaround suggested here. But that is far from optimal as the quotes show in the edition name in Plex.
